llvm/lib/Target/SPIRV/SPIRVStructurizer.cpp

```diff
@@ -87,6 +87,8 @@ BasicBlock *getExitFor(const ConvergenceRegion *CR) {
 // Returns the merge block designated by I if I is a merge instruction, nullptr
 // otherwise.
 BasicBlock *getDesignatedMergeBlock(Instruction *I) {
+  if (I == nullptr)
+    return nullptr;
```

Comment:
Would `dyn_cast_or_null<IntrinsicInst>(I)` do the trick to avoid adding this check?

Reply:
done